src/logic/basic.lean

```diff
@@ -797,6 +797,14 @@ by simp only [or_imp_distrib, forall_and_distrib, forall_eq]
 @[simp] theorem exists_eq_right {a' : α} : (∃ a, p a ∧ a = a') ↔ p a' :=
 (exists_congr $ by exact λ a, and.comm).trans exists_eq_left
 
+@[simp] lemma exists_eq_right_right {p : α → Prop} {q : Prop} {x : α} :
+  (∃ (y : α), p y ∧ q ∧ y = x) ↔ p x ∧ q :=
+⟨λ ⟨_, hp, hq, rfl⟩, ⟨hp, hq⟩, λ ⟨hp, hq⟩, ⟨x, hp, hq, rfl⟩⟩
+
+@[simp] lemma exists_eq_right_right' {p : α → Prop} {q : Prop} {x : α} :
+  (∃ (y : α), p y ∧ q ∧ x = y) ↔ p x ∧ q :=
+⟨λ ⟨_, hp, hq, rfl⟩, ⟨hp, hq⟩, λ ⟨hp, hq⟩, ⟨x, hp, hq, rfl⟩⟩
```

Comment:
For consistency with the lines above these should probably be `lemma`. Also, you don't need `p` or `q` here, I think they're already defined in a `variables` above. Finally, I think it would make sense to match the names above:

```suggestion
@[simp] theorem exists_eq_right_right {a' : α} :
  (∃ (a : α), p a ∧ q ∧ a = a') ↔ p a' ∧ q :=
⟨λ ⟨_, hp, hq, rfl⟩, ⟨hp, hq⟩, λ ⟨hp, hq⟩, ⟨a', hp, hq, rfl⟩⟩

@[simp] theorem exists_eq_right_right' {x : α} :
  (∃ (a : α), p a ∧ q ∧ a' = a) ↔ p a' ∧ q :=
⟨λ ⟨_, hp, hq, rfl⟩, ⟨hp, hq⟩, λ ⟨hp, hq⟩, ⟨a', hp, hq, rfl⟩⟩
```

(untested!)